src/Groups/perm.jl

```diff
@@ -387,3 +387,168 @@ function cycle_structure(g::PermGroupElem)
     # TODO: use SortedDict from DataStructures.jl ?
     return Pair{Int,Int}[ i+1 => c[i] for i in 1:length(c) if GAP.Globals.ISB_LIST(c, i) ]
 end
+
+
+# The following code implements a new way to input permutations in Julia. For example
+# it is possible to create a permutation as follow
+# pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+# For this we use macros to modify the syntax tree of (1,2,3)(4,5)(6,7,8) such that
+# Julia can deal with the expression.
+
+
+################################################################################
+#
+#   perm
+#
+@doc Markdown.doc"""
+    @perm(ex)
+Macro to input a permutation as
+`pi = @perm (1,2,3)(4,5)(6,7,8)` to obtain
+the permutation `(1,2,3)(4,5)(6,7,8)`, that is, the output of
+`cperm([1,2,3],[4,5],[6,7,8])`.
+# Examples
+```jldoctest
+julia> @perm (1,2,3)(4,5)(6,7,8)
+(1,2,3)(4,5)(6,7,8)
+```
+"""
+macro perm(ex)
+    res = []
+    
+    if typeof(ex) != Expr
+        error("Input is not a permutation.")
+    end
+    
+    if ex.head != :call && ex.head != :tuple
+        error("Input is not a permutation.")
+    end
+   
+    while ex.head == :call
+        pushfirst!(res, Expr(:vect, ex.args[2:end]...))
+        ex = ex.args[1]
+        if typeof(ex) != Expr
+            error("Input is not a permutation.")
+        end
+        if ex.head != :call && ex.head != :tuple
+            error("Input is not a permutation.")
+        end
+    end
+    
+    if ex.head != :tuple
+        error("Input is not a permutation.")
+    end
+    
+    pushfirst!(res, Expr(:vect,ex.args...))
+
+    return esc(:(Oscar.cperm($(res...))))
+end
+
+
+################################################################################
+#
+#   perm(n,gens)
+#
+@doc Markdown.doc"""
+    @perm(n,gens)
+Macro to input a list of permutations which are generated as elements of
+the `symmetric_group(n)` with the function `cperm`.
+# Examples
+```jldoctest
+julia> gens = Oscar.Permutations.@perm 14 [
```

Comment:
`Oscar.Permutations` does not exist
```suggestion
julia> gens = @perm 14 [
```